### PR TITLE
[DONE] 5.2.3

### PR DESCRIPTION
**Pull request by xHyroM**
**Features**
---
- [x] Cleanup
- [x] Better jsdoc for docs
- [x] Command classes
- [x] Cooldown bypass for app owner/team
- [x] Cooldown fix
- [x] Typings for command classes
- [x] Event classes
- [x] Typings for event classes
- [x] Workflows (auto checks for PRs)
- [x] Min node version in package.json
- [x] GPayload
- [x] GPayload typings
- [x] Sending only attachments in message
- [x] Sending only embeds in message
- [x] Back Structures with djs v13
- [x] Removed `guild.prefix/language` (use `getCommandPrefix(), getLanguage()`)
- [x] Added `.setDisabled()` to `MessageSelectMenu`
- [x] Support `ws`(raw) events
- [x] client.events -> client.gevents
- [x] interaction.think() in `slash cmd`
- [x] interaction.createdTimestamp
- [x] GInteraction typings
- [x] GInteraction jsdoc
- [x] GInteraction
- [x] GPayload test
- [x] Normal arguments
- [x] expectedArgs => args
- [x] SlashCommand.STRING => ArgumentType.STRING
- [x] GEvents to @gcommands/events
- [x] Case sensitive commands
- [x] Embed attachments fix
- [x] Major fixes
- [x] Final Testing

**Website**
- [x] Guide gcommands.js.org/guide
- [x] Docs gcommands.js.org/docs
- [x] gcommands.js.org => gcommands.js.org/docs(Auto redirect to docs)

**Status**
---
- [x] These changes have been tested and formatted properly.
- [x] This PR introduces some Breaking changes.

### CODE REVIEW

**Review by iZboxo — APPROVED** [review]

**Review by SebestikCZ — APPROVED** [review]
Fine

**Review by Gabe616 — APPROVED** [review]
pog

File: src/base/GCommands.js [hunk 1]
```diff
@@ -57,19 +57,18 @@ class GCommands extends GCommandsBase {
...
+        */
+        this.shardClusterName = options.shardClusterName || "shard";
 
         /**
          * database
          * @property {Object} database
         */
         this.database = options.database || undefined;
 
-        this.client.categories = fs.readdirSync("./" + this.cmdDir );
```
Comment (SebestikCZ) [inline src/base/GCommands.js:72]: Get back categories!

**Review by SebestikCZ — CHANGES_REQUESTED** [review]
Hyro, on 240 line in GMessage.js there is an conflicting expression, replace updater.checkDjsVersion("13") with ifDjsV13!

File: src/base/GCommands.js (hunk 1 above)
Comment (xHyroM) [reply src/base/GCommands.js:72]: Okay, but I'll rename to gcategories.

**Comment by xHyroM** [comment]
> 
> 
> Hyro, on 240 line in GMessage.js there is an conflicting expression, replace updater.checkDjsVersion("13") with ifDjsV13!

resolved

**Review by SebestikCZ — APPROVED** [review]
Thanks, and now check your DMs

**Review by hernikplays — APPROVED** [review]

File: src/managers/GEventHandling.js [hunk 2]
```diff
@@ -546,33 +518,21 @@ class GEventHandling {
...
-                data.components = result.components;
-            }
-            if(typeof result == "object" && result.embeds) {
-                if(!Array.isArray(result.embeds)) result.embeds = [result.embeds]
-                data.embeds = result.embeds;
-            }
-            if(result.embeds && !result.content) result.content = "\u200B"
+            let GPayloadResult = GPayload.create(channel, result)
+                .resolveData()
+                .resolveFiles();
```
Comment (xHyroM) [inline src/managers/GEventHandling.js:532]:
```suggestion
```

Can't edit attachments.

File: src/managers/GEventHandling.js [hunk 3]
```diff
@@ -546,33 +518,21 @@ class GEventHandling {
...
             
             let apiMessage = (await this.client.api.webhooks(this.client.user.id, interaction.token).messages[result.messageId ? result.messageId : "@original"].patch({
-                data,
-                files: finalFiles   
+                data: GPayloadResult.data,
+                files: GPayloadResult.files
```
Comment (xHyroM) [inline src/managers/GEventHandling.js:535]:
```suggestion
```

Can't edit attachments.

File: src/structures/GMessage.js [hunk 4]
```diff
@@ -66,46 +38,12 @@ if(!ifDjsV13) {
...
-                        finalFiles.push({
-                            attachment: file.attachment,
-                            name: file.name,
-                            file: file.attachment
-                        })
-                    })
-                }
+                let GPayloadResult = GPayload.create(this.channel, result)
+                    .resolveData()
+                    .resolveFiles();
```
Comment (xHyroM) [inline src/structures/GMessage.js:43]:
```suggestion
```

Can't edit attachments.

File: src/structures/GMessage.js [hunk 5]
```diff
@@ -115,33 +53,21 @@ if(!ifDjsV13) {
...
-                    data.components = result.components;
-                }
-                if(typeof result == "object" && result.embeds) {
-                    if(!Array.isArray(result.embeds)) result.embeds = [result.embeds]
-                    data.embeds = result.embeds;
-                }
-                if(result.embeds && !result.content) result.content = "\u200B"
+                let GPayloadResult = GPayload.create(this.channel, result)
+                    .resolveData()
+                    .resolveFiles();
```
Comment (xHyroM) [inline src/structures/GMessage.js:58]:
```suggestion
```

Can't edit attachments.

File: src/structures/GMessage.js [hunk 6]
```diff
@@ -163,26 +89,14 @@ if(!ifDjsV13) {
...
-                }
-                if(typeof result == "object" && result.embeds) {
-                    if(!Array.isArray(result.embeds)) result.embeds = [result.embeds]
-                    data.embeds = result.embeds;
-                }
-                if(result.embeds && !result.content) result.content = "\u200B"
-
+                let GPayloadResult = GPayload.create(this.channel, result)
+                    .resolveData()
+                    .resolveFiles();
```
Comment (xHyroM) [inline src/structures/GMessage.js:94]:
```suggestion
```

Can't edit attachments.

File: src/structures/GMessage.js [hunk 7]
```diff
@@ -313,46 +199,14 @@ if(!ifDjsV13) {
...
-                finalFiles.push({
-                    attachment: file.attachment,
-                    name: file.name,
-                    file: file.attachment
-                })
-            })
-        }
+        let GPayloadResult = GPayload.create(this.channel, result)
+            .resolveData()
+            .resolveFiles();
```
Comment (xHyroM) [inline src/structures/GMessage.js:205]:
```suggestion
```

Can't edit attachments.

File: src/structures/GMessage.js [hunk 8]
```diff
@@ -313,46 +199,14 @@ if(!ifDjsV13) {
...
-            data: {
-                allowed_mentions: options.allowedMentions,
-                content: content,
-                components: options.components,
-                options,
-                embed: embed || null
-            },
-            files: finalFiles
+            data: GPayloadResult.data,
+            files: GPayloadResult.files
```
Comment (xHyroM) [inline src/structures/GMessage.js:209]:
```suggestion
```

Can't edit attachments.

File: src/structures/InteractionEvent.js [hunk 9]
```diff
@@ -312,33 +284,21 @@ class InteractionEvent {
     }
 
     async slashEdit(result, update) {
-        var data = {}
+        let GPayloadResult = GPayload.create(this.channel, result)
+            .resolveData()
+            .resolveFiles();
```
Comment (xHyroM) [inline src/structures/InteractionEvent.js:290]:
```suggestion
```

Can't edit attachments.

**Review by WurstMod — APPROVED** [review]

**Comment by xHyroM** [comment]
https://github.com/Garlic-Team/GCommands/issues/39

**Comment by xHyroM** [comment]
Preparation for normal arguments.
```js
args: [
 // etc (same as slash args) https://gcommands.js.org/guide/guide/beginner/arguments.html#without-slash-group
]
```

File: src/base/GCommands.js [hunk 10]
```diff
@@ -1,121 +1,164 @@
...
+         * @type {Client}
+        */
         this.client = client;
 
+        /**
+         * caseSensitiveCommands
+         * @type {Boolean}
+         * @default true
+        */
+        this.caseSensitiveCommands = Boolean(options.caseSensitiveCommands) || true
```
Comment (xHyroM) [inline src/base/GCommands.js:40]: Not working, need fix

**Comment by iZboxo** [comment]
I think everything is corect.

**Review by iZboxo — APPROVED** [review]
Everything is correct. Gcommands is best.

**Review by Gabe616 — APPROVED** [review]

**Pull request merged**